File: code/126/solution.py
from random import random
from math import *


def gen(a, b, c, layer):
    return 2 * a * b + 2 * b * c + 2 * a * c + 4 * (layer - 1) * (a + b + c + layer - 2)


upper = 20000
cnt = [0] * upper

i = 1
while gen(i, i + 1, i + 2, 1) < upper:
    j = i
    while gen(i, j, j + 1, 1) < upper:
        k = j
        while gen(i, j, k, 1) < upper:
            layer = 1
            while gen(i, j, k, layer) < upper:
                cnt[gen(i, j, k, layer)] += 1
                layer += 1
            k += 1
        j += 1
    i += 1

for i in range(1, upper):
    if cnt[i] == 1000:
        print(i)
        break

File: code/126/solution.cpp
#include "../template/bigint.cpp"
#include "../template/china_reminder.cpp"
#include "../template/euler.cpp"
#include "../template/pell.cpp"
#include "../template/prime.cpp"
#include "../template/prime_cnt.cpp"
#include <iostream>/*{{{*/
#include <map>
#include <math.h>
#include <set>
#include <vector>
const long long e2 = 100;
const long long e3 = e2 * 10;
const long long e4 = e3 * 10;
const long long e5 = e4 * 10;
const long long e6 = e4 * e2;
const long long e7 = e6 * 10;
const long long e8 = e7 * 10;
const long long e9 = e8 * 10;
long long _mod     = e9 + 7;
using namespace std; /*}}}*/

int gen(int a, int b, int c, int layer) { return 2 * a * c + 2 * b * c + 2 * a * b + 4 * (layer - 1) * (a + b + c + layer - 2); }
const int upper = 20000;
int cnt[upper];
int main() {
    for (int i = 1; gen(i, i + 1, i + 2, 1) < upper; i++) {
        for (int j = i; gen(i, j, j + 1, 1) < upper; j++) {
            for (int k = j; gen(i, j, k, 1) < upper; k++) {
                for (int layer = 1; gen(i, j, k, layer) < upper; layer++) {
                    cnt[gen(i, j, k, layer)]++;
                }
            }
        }
    }
    for (int i = 1; i < upper; i++) {
        if (cnt[i] == 1000) {
            cout << i << endl;
            break;
        }
    }
    return 0;
}
